### PR TITLE
🦖 🦖 🦖 [DOMA-2851] - B2C Apps 

### PR DESCRIPTION
**Pull request by SavelevMatthew**

### CODE REVIEW

File: apps/condo/domains/miniapp/schema/B2CApp.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                            id: resolvedData[fieldPath],
+                        })
+                        if (!validBuild) {
+                            return addFieldValidationError(RESTRICT_BUILD_SELECT_ERROR)
+                        }
+                    }
+                },
+            },
+        },
+        builds: {
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/B2CApp.js:60]: Please remove this field. It's too heavy.

File: apps/condo/domains/miniapp/schema/B2CApp.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                        app: { id_not: appId },
+                        deletedAt: null,
+                    })
+                    if (restrictedBuilds.length) {
+                        return addFieldValidationError(RESTRICT_BUILD_APP_CHANGE_ERROR)
+                    }
+                },
+            },
+        },
+        properties: {
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/B2CApp.js:80]: Please, don't use m2m. It's too heavy

File: apps/condo/domains/miniapp/schema/B2CApp.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                        app: { id_not: appId },
+                        deletedAt: null,
+                    })
+                    if (restrictedProperties.length) {
+                        return addFieldValidationError(RESTRICT_PROPERTY_APP_CHANGE_ERROR)
+                    }
+                },
+            },
+        },
+        accessRights: {
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/B2CApp.js:100]: It's better to give access to this field only for admins. It's access

File: apps/condo/domains/miniapp/schema/B2CAppProperty.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            isRequired: false,
+            kmigratorOptions: { null: true, on_delete: 'models.PROTECT' },
+            access: {
+                update: access.canUpdateAppLink,
+            },
+            hooks: {
+                validateInput: async ({ operation, resolvedData, addFieldValidationError }) => {
+                    if (operation === 'update') {
+                        if (resolvedData.hasOwnProperty('app') && resolvedData.app !== null) {
+                            return addFieldValidationError(RESTRICT_PROPERTY_APP_CHANGE_ERROR)
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/B2CAppProperty.js:44]: it' look like you already check it by access?

File: apps/condo/domains/miniapp/schema/B2CAppProperty.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                    if (suggestionAddress.toLowerCase() !== inputAddress.toLowerCase()) {
+                        return addFieldValidationError(INCORRECT_ADDRESS_ERROR)
+                    }
+                },
+            },
+        },
+    },
+    hooks: {
+        resolveInput: async ({ resolvedData }) => {
+            if (resolvedData && resolvedData.hasOwnProperty('app') && resolvedData['app'] === null) {
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/B2CAppProperty.js:82]: it look like a hack for admin interface restrictions

File: apps/condo/domains/miniapp/schema/B2CAppProperty.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        update: access.canManageB2CAppProperties,
+        delete: false,
+        auth: true,
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['address', 'app'],
+                condition: 'Q(deletedAt__isnull=True)',
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/B2CAppProperty.js:106]: 🥇

File: apps/condo/domains/miniapp/schema/fields/accessRight.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+const SERVICE_USER_FIELD = {
+    schemaDoc: `Link to user. Note, that user must be of "${SERVICE}" type`,
+    type: Relationship,
+    ref: 'User',
+    isRequired: true,
+    knexOptions: { isNotNullable: true }, // Required relationship only!
+    kmigratorOptions: { null: false, on_delete: 'models.PROTECT' },
+    hooks: {
+        validateInput: async ({ resolvedData, fieldPath, addFieldValidationError }) => {
```
Comment (pahaz) [inline apps/condo/domains/miniapp/schema/fields/accessRight.js:14]: can we add the same check for other apps?

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/miniapp/schema/B2CApp.js (hunk 1 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/B2CApp.js]: removed

File: apps/condo/domains/miniapp/schema/B2CApp.js (hunk 3 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/B2CApp.js:58]: removed

File: apps/condo/domains/miniapp/schema/B2CApp.js (hunk 2 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/B2CApp.js]: removed

File: apps/condo/domains/miniapp/schema/B2CAppProperty.js (hunk 5 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/B2CAppProperty.js]: removed

File: apps/condo/domains/miniapp/schema/B2CAppProperty.js (hunk 4 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/B2CAppProperty.js]: was admin panel hack. removed

File: apps/condo/domains/miniapp/schema/fields/accessRight.js (hunk 7 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/fields/accessRight.js:14]: It's already integrated with B2B Apps as well...

Will add it to billing / acquiring too

Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/fields/accessRight.js:14]: done

File: apps/condo/domains/miniapp/schema/B2CApp.js (hunk 3 above)
Comment (SavelevMatthew) [reply apps/condo/domains/miniapp/schema/B2CApp.js:58]: You can't fetch any accessRIght if you're not admin / support....And specifying additional access here will not remove it from GQL types as well as not improve query performance....So, I'll left it as it was

**Pull request merged**